### PR TITLE
Add custom Jayvee formatter

### PR DESCRIPTION
**Pull request by georg-schwarz**
Adds a custom formatter to via the language server protocol.
Future updates might enable using the formatted from CLI.

Works around an issue with comments not being correctly formatted: https://github.com/eclipse-langium/langium/issues/1351

### CODE REVIEW

File: example/electric-vehicles.jv [hunk 1]
```diff
@@ -112,32 +114,80 @@ pipeline ElectricVehiclesPipeline {
 // that this value type builts on. User-defined value types always place additional constraints on existing value types.
 valuetype VehicleIdentificationNumber10 oftype text {
 	// 10. Value types can be further refined by providing constraints.
-	constraints: [
+	constraints : [
```
Comment (joluj) [inline example/electric-vehicles.jv:117]: This should not happen

File: example/workbooks-xlsx.jv [hunk 2]
```diff
@@ -32,8 +30,16 @@ pipeline LightTrappingSiliconSolarCellsPipeline {
 	}
 
 	block NameHeaderWriter oftype CellWriter {
-		at: range F1:L1;
-		write: ["F","G","nm","wl","n2", "k2", "alpha (cm-1)2"];
+		at: range F1: L1;
```
Comment (joluj) [inline example/workbooks-xlsx.jv:33]:
```suggestion
		at: range F1:L1;
```

**Review by joluj — APPROVED** [review]
- `cars.jv` uses tabs, `gtfs-rt.jv` uses spaces. I'd unify this.
  - Personally, I'm team 2 spaces but I don't really mind. 
- Every file should end with exactly one empty line https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline
- Some formatting errors (see comments in files)

These points can be tackled in other PRs though. But we should open issues at least.

**Comment by georg-schwarz** [comment]
Thx, good catches!

I noticed in the meanwhile that Langium itself does AST-node-specific formatting. Probably sth we should consider as well (e.g., for the colons).
https://github.com/eclipse-langium/langium/blob/main/packages/langium/src/grammar/lsp/grammar-formatter.ts

**Comment by georg-schwarz** [comment]
Fixed the issues and converted all examples to use 2 spaces.

> Every file should end with exactly one empty line https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

That should be tackled in a separate issue.

In general, I'm not too happy with the implementation. At some points two formatting are valid that the formatting function switches between when applied. But I'll leave that to future PRs, together with the refactoring to apply formatting not globally but AST node specifically.

**Review by joluj — APPROVED** [review]

**Pull request merged**